File: 02_ProgrammingFundamentals/_02_DataTypesAndVariables/_02_Exercise/_11_Snowballs.java
package _02_ProgrammingFundamentals._02_DataTypesAndVariables._02_Exercise;

import java.util.Scanner;

public class _11_Snowballs {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        // Read input value
        int numberOfSnowballs = Integer.parseInt(scan.nextLine());

        // Add variables
        double max = Double.NEGATIVE_INFINITY;
        int maxSnowballSnow = 0;
        int maxSnowballTime = 0;
        int maxSnowballQuality = 0;
        double maxSnowballValue = 0;

        // Determine which one is the best snowball
        for (int i = 0; i < numberOfSnowballs; i++) {
            int snowballSnow = Integer.parseInt(scan.nextLine());
            int snowballTime = Integer.parseInt(scan.nextLine());
            int snowballQuality = Integer.parseInt(scan.nextLine());
            double snowballValue = Math.pow((1.00 * snowballSnow / snowballTime), 1.00 * snowballQuality);
            
            if (snowballValue > max){
                maxSnowballSnow = snowballSnow;
                maxSnowballTime = snowballTime;
                maxSnowballQuality = snowballQuality;
                maxSnowballValue = snowballValue;
                max = snowballValue;
            }
        }
        
        // Print result
        System.out.printf("%d : %d = %.0f (%d)", maxSnowballSnow, maxSnowballTime, maxSnowballValue, maxSnowballQuality);
        
        scan.close();
    }
}

File: 24_C#Fundamentals/_02_DataTypesAndVariables/_02_Exercise/_11_Snowballs/_11_Snowballs.cs
using System;
using System.Numerics;

public class _11_Snowballs
{
    public static void Main(string[] args)
    {
        // Read input value
        int numberOfSnowballs = int.Parse(Console.ReadLine());

        // Add variables
        BigInteger max = 0;
        int maxSnowballSnow = 0;
        int maxSnowballTime = 0;
        int maxSnowballQuality = 0;
        BigInteger maxSnowballValue = 0;

        // Determine which one is the best snowball
        for (int i = 0; i < numberOfSnowballs; i++)
        {
            int snowballSnow = int.Parse(Console.ReadLine());
            int snowballTime = int.Parse(Console.ReadLine());
            int snowballQuality = int.Parse(Console.ReadLine());
            BigInteger snowballValue = BigInteger.Pow((snowballSnow / snowballTime), snowballQuality);

            if (snowballValue > max)
            {
                maxSnowballSnow = snowballSnow;
                maxSnowballTime = snowballTime;
                maxSnowballQuality = snowballQuality;
                maxSnowballValue = snowballValue;
                max = snowballValue;
            }
        }

        // Print result
        Console.WriteLine($"{maxSnowballSnow} : {maxSnowballTime} = {maxSnowballValue:F0} ({maxSnowballQuality})");
    }
}
